packages/@aws-cdk/aws-rds/lib/serverless-cluster.ts

```diff
@@ -584,3 +608,79 @@ class ImportedServerlessCluster extends ServerlessClusterBase implements IServer
     return this._clusterReadEndpoint;
   }
 }
+
+/**
+ * Properties for ``ServerlessClusterFromSnapshot``
+ */
+export interface ServerlessClusterFromSnapshotProps extends ServerlessClusterBaseProps {
+  /**
+   * The identifier for the DB instance snapshot or DB cluster snapshot to restore from.
+   * You can use either the name or the Amazon Resource Name (ARN) to specify a DB cluster snapshot.
+   * However, you can use only the ARN to specify a DB instance snapshot.
+   */
+  readonly snapshotIdentifier: string;
+
+  /**
+   * Master user credentials.
+   *
+   * Note - It is not possible to change the master username for a snapshot;
+   * however, it is possible to provide (or generate) a new password.
+   *
+   * @default - The existing username and password from the snapshot will be used.
+   */
+  readonly credentials?: SnapshotCredentials;
+}
+
+/**
+ * A Aurora Serverless Cluster restored from a snapshot.
+ *
+ * @resource AWS::RDS::DBCluster
+ */
+export class ServerlessClusterFromSnapshot extends ServerlessClusterNew {
+  public readonly clusterIdentifier: string;
+  public readonly clusterEndpoint: Endpoint;
+  public readonly clusterReadEndpoint: Endpoint;
+  protected enableDataApi?: boolean;
```

Comment:
Can we move this up to `ServerlessClusterNew`?

Reply:
`enableDataApi` --> done

If I move the `readonly` fields to `ServerlessClusterNew`, it shows message `Property 'clusterIdentifier' has no initializer and is not definitely assigned in the constructor.`. I could declare them as `abstract`. However, then I still have to create those fields in the sub classes.